pkg/hooks/loader.go

```diff
@@ -225,14 +244,34 @@ func (h *Hook) GetConfigMocks() ([]*models.Mock, error) {
 
 func (h *Hook) DeleteTcsMock(mock *models.Mock) bool {
 	isDeleted := h.tcsMocks.delete(mock.TestModeInfo)
+	h.matchedMocks = append(h.matchedMocks, mock.Name)
+	h.persistentMatchedMocks[mock] = true
 	return isDeleted
 }
 
 func (h *Hook) DeleteConfigMock(mock *models.Mock) bool {
 	isDeleted := h.configMocks.delete(mock.TestModeInfo)
+	h.matchedMocks = append(h.matchedMocks, mock.Name)
+	h.persistentMatchedMocks[mock] = true
 	return isDeleted
 }
 
```

Comment:
incase if the mock isn't deleted (meaning isn't matched) then also its gets pushed in the map for delete, and update.

We should only put the mock that's matched right?

Reply:
delete functions are used when a match takes place 
https://github.com/keploy/keploy/blob/e117ea1ff0113a408fb73a0c27a9e8404b0d825a/pkg/proxy/integrations/grpcparser/request_to_response_mapping.go#L76
https://github.com/keploy/keploy/blob/e117ea1ff0113a408fb73a0c27a9e8404b0d825a/pkg/proxy/integrations/httpparser/match.go#L75
https://github.com/keploy/keploy/blob/e117ea1ff0113a408fb73a0c27a9e8404b0d825a/pkg/proxy/integrations/mongoparser/match.go#L55

Reply:
```func (db *treeDb) delete(key interface{}) bool {
	db.mutex.Lock()
	defer db.mutex.Unlock()
	_, found := db.rbt.Get(key)
	if !found {
		return false
	}
	db.rbt.Remove(key)
	return true
}
```

```isDeleted := h.DeleteTcsMock(mock)
		if !isDeleted {
			continue
		}
```


If some other test case matched to the mock , it will be deleted from memory , isdeleted for this mock would be false and we will continue looking for next most suitable mock. Incase we go with overwriting without checking is deleted, we will be triggering map related functions multiple time

Reply:
any update on this?

Reply:
> Incase we go with overwriting without checking is deleted, we will be triggering map related functions multiple time

couldn't understand this

Reply:
mock match doesn't guarantee mock being deleted. If some other call got matched with the best matched mock it will be deleted (eg: we're looking for a call-1 , mock5 is best match but it's already used by call-5 , so it will be removed from the tree, in that can call-1 will look for another mock.

Incase if we ignore the isdeleted var we would be doing map[mock5] = true twice. And this map operation takes more time complexity, so better to trigger map[mock5] = true only once.

Reply:
i have made the changes. please have a look